src/OpenTelemetry.AutoInstrumentation.Loader/AssemblyResolver.NetFramework.cs

```diff
@@ -50,33 +41,91 @@ private static string ResolveManagedProfilerDirectory()
             try
             {
                 var mongoAssembly = Assembly.Load(assemblyName);
-                logger.Debug<string, bool>("Assembly.Load(\"{0}\") succeeded={1}", assemblyName, mongoAssembly != null);
+                _logger.Debug<string, bool>("Assembly.Load(\"{0}\") succeeded={1}", assemblyName, mongoAssembly != null);
                 return mongoAssembly;
             }
             catch (Exception ex)
             {
-                logger.Debug(ex, "Assembly.Load(\"{0}\") Exception: {1}", assemblyName, ex.Message);
+                _logger.Debug(ex, "Assembly.Load(\"{0}\") Exception: {1}", assemblyName, ex.Message);
             }
 
             return null;
         }
 
-        var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName}.dll");
+        var path = Path.Combine(Path.GetDirectoryName(_managedProfilerDirectory) ?? _managedProfilerDirectory, $"{assemblyName}.dll");
+        if (!File.Exists(path))
+        {
+            path = Path.Combine(_managedProfilerDirectory, $"{assemblyName}.dll");
+            if (!File.Exists(path))
+            {
+                var link = Path.Combine(_managedProfilerDirectory, $"{assemblyName}.dll.link");
+                if (File.Exists(link))
+                {
+                    try
+                    {
+                        var linkPath = File.ReadAllText(link).Trim();
+                        path = Path.Combine(Path.GetDirectoryName(_managedProfilerDirectory) ?? _managedProfilerDirectory, linkPath, $"{assemblyName}.dll");
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.Debug(ex, "Error reading .link file {0}", link);
+                    }
+                }
+                else
+                {
+                    // Not found
+                    return null;
+                }
+            }
+        }
+
         if (File.Exists(path))
         {
             try
             {
                 var loadedAssembly = Assembly.LoadFrom(path);
-                logger.Debug<string, bool>("Assembly.LoadFrom(\"{0}\") succeeded={1}", path, loadedAssembly != null);
+                _logger.Debug<string, bool>("Assembly.LoadFrom(\"{0}\") succeeded={1}", path, loadedAssembly != null);
                 return loadedAssembly;
             }
             catch (Exception ex)
             {
-                logger.Debug(ex, "Assembly.LoadFrom(\"{0}\") Exception: {1}", path, ex.Message);
+                _logger.Debug(ex, "Assembly.LoadFrom(\"{0}\") Exception: {1}", path, ex.Message);
             }
         }
 
         return null;
     }
+
+    [DllImport("OpenTelemetry.AutoInstrumentation.Native.dll")]
+    private static extern int GetDetectedNetFrameworkVersion();
```

Comment:
Should we add a comment here that indicates that it will only detect versions up to 4.7.2?